src/magic_folder/test/test_upload.py

```diff
@@ -0,0 +1,164 @@
+import io
+import attr
+from tempfile import mktemp
+
+from testtools.matchers import (
+    AfterPreprocessing,
+    Always,
+)
+from testtools.twistedsupport import (
+    succeeded,
+)
+from hypothesis import (
+    given,
+)
+from hypothesis.strategies import (
+    binary,
+)
+from twisted.python.filepath import (
+    FilePath,
+)
+from hyperlink import (
+    DecodedURL,
+)
+from ..magic_folder import (
+    UploaderService,
+)
+from magic_folder.snapshot import (
+    create_local_author,
+    create_snapshot,
+)
+from twisted.internet import reactor
+from .common import (
+    SyncTestCase,
+)
+from .strategies import (
+    path_segments,
+)
+from eliot import (
+    Message,
+)
+from magic_folder.testing.web import (
+    create_fake_tahoe_root,
+    create_tahoe_treq_client,
+)
+from magic_folder.tahoe_client import (
+    create_tahoe_client,
+)
+from allmydata.uri import is_uri
+
+@attr.s
+class MemorySnapshotStore(object):
```

Comment:
It would be convenient if we didn't need this.  I usually consider a ":memory:"-connected SQLite3 database to be Good Enough™ for use in tests.  It doesn't do real I/O so it's not slow like really using the database would be.  It's pretty opaque which can be annoying but with a simple CRUD-ish interface like this one I don't think that's going to be much of an issue.  Test doubles are great but avoiding them is even better. :)

Reply:
I keep forgetting this..

Reply:
Looks like, with the new config changes, it is going to be hard to do this (without modifying the current config code..) because the filename is hard coded into the code and is not parameterized anymore. (Overall, that seem like a good decision though)